### DISCUSSION TITLE
Copy to highlight how can we implement this

### DISCUSSION THREAD

**Discussion by Daszbin** [cefsharp/CefSharp#4269] [General]
I was wondering how can we implement copy link to highlight through the contextmenuhandler and is their an API in CEF to implement this feature.

**Comment by amaitland** [comment]
>  is their an API in CEF to implement this feature.

The API doc is available at http://cefsharp.github.io/api/105.3.x/html/T_CefSharp_IContextMenuHandler.htm

> copy link to highlight

What exactly are you trying to achieve? What is copy link to highlight exactly?

**Reply by Daszbin** [reply]
Share or link to quotes & text in Chrome
You can create a shareable link that goes directly to specific highlighted text on a page.

Computer AndroidiPhone & iPad
To create a link that opens directly to highlighted text: 

On your computer, open Chrome. 
Go to a page with text you want to share. 
To highlight the text you want to share, click and hold, then drag your mouse.
To open the context menu, right-click on the highlighted text.
Select Copy link to highlight. 
If you can’t select this option, this feature may not work for the selected content.
Paste the link anywhere, like an email or message thread.
Tip: To remove the highlight from the text in the linked content, right-click the highlighted text and select Remove highlight.

**Reply by amaitland** [reply]
You can access the `SelectedText` via `IContextMenuParams`.

As for exactly what `Chromium` is doing you can search the source code online via https://source.chromium.org/chromium or you can just workout whatever behaviour suites your needs.

There are some examples of context menus in the project source
- https://github.com/cefsharp/CefSharp/blob/cefsharp/106/CefSharp.Wpf/Handler/ContextMenuHandler.cs
- https://github.com/cefsharp/CefSharp/blob/cefsharp/106/CefSharp.WinForms.Example/Handlers/MenuHandler.cs
